### ISSUE TITLE
add support for more of PyTorch's class-based modules

### ISSUE THREAD

**Issue by shaymeister** [shaymeister/parcae#4]
# Convolution Layers
- [ ] torch.nn.modules.ConvTranspose1d
- [ ] torch.nn.modules.ConvTranspose2d
- [ ] torch.nn.modules.ConvTranspose3d
- [ ] torch.nn.modules.LazyConv1d
- [ ] torch.nn.modules.LazyConv2d
- [ ] torch.nn.modules.LazyConv3d
- [ ] torch.nn.modules.LazyConvTranspone1d
- [ ] torch.nn.modules.LazyConvTranspose2d
- [ ] torch.nn.modules.LazyConvTranspose3d

# Pooling Layers
- [ ] torch.nn.modules.MaxUnpool1d
- [ ] torch.nn.modules.MaxUnpool2d
- [ ] torch.nn.modules.MaxUnpool3d
- [ ] torch.nn.modules.FractionalMaxPool2d
- [ ] torch.nn.modules.FractionalMaxPool3d
- [ ] torch.nn.modules.LPPool1d
- [ ] torch.nn.modules.LPPool2d

# Padding Layers
- [ ] torch.nn.modules.ReflectionPad1d
- [ ] torch.nn.modules.ReflectionPad2d
- [ ] torch.nn.modules.ReplicationPad1d
- [ ] torch.nn.modules.ReplicationPad2d
- [ ] torch.nn.modules.ReplicationPad3d
- [ ] torch.nn.modules.ZeroPad2d
- [ ] torch.nn.modules.ConstantPad1d
- [ ] torch.nn.modules.ConstantPad2d
- [ ] torch.nn.modules.ConstantPad3d

# Non-linear Activations (weighted sum, nonlinearity)
- [ ] torch.nn.modules.Hardshrink
- [ ] torch.nn.modules.Hardsigmoid
- [ ] torch.nn.modules.Hardtanh
- [ ] torch.nn.modules.Hardswish
- [ ] torch.nn.modules.LogSigmoid
- [ ] torch.nn.modules.MultiheadAttention
- [ ] torch.nn.modules.RReLU
- [ ] torch.nn.modules.SELU
- [ ] torch.nn.modules.CELU
- [ ] torch.nn.modules.GELU
- [ ] torch.nn.modules.Sigmoid
- [ ] torch.nn.modules.SiLU
- [ ] torch.nn.modules.Mish
- [ ] torch.nn.modules.Softplus
- [ ] torch.nn.modules.Softshrink
- [ ] torch.nn.modules.Softsign
- [ ] torch.nn.modules.Tanh
- [ ] torch.nn.modules.Tanhshrink
- [ ] torch.nn.modules.Threshold

# Non-linear Activations (other)
- [ ] torch.nn.modules.Softmin
- [ ] torch.nn.modules.Softmax
- [ ] torch.nn.modules.Softmax2d
- [ ] torch.nn.modules.LogSoftmax
- [ ] torch.nn.modules.AdaptiveLogSoftmaxWithLoss

# Normalization Layers
- [ ] torch.nn.modules.LazyBatchNorm1d
- [ ] torch.nn.modules.LazyBatchNorm2d
- [ ] torch.nn.modules.LazyBatchNorm3d
- [ ] torch.nn.modules.SyncBatchNorm
- [ ] torch.nn.modules.LayerNorm
- [ ] torch.nn.modules.LocalResponseNorm

# Transformer Layers
- [ ] torch.nn.modules.Transformer
- [ ] torch.nn.modules.TransformerEncoder
- [ ] torch.nn.modules.TransformerDecoder
- [ ] torch.nn.modules.TransformerEncoderLayer
- [ ] torch.nn.modules.TransformerDecoderLayer

# Linear Layers
- [ ] torch.nn.modules.Bilinear
- [ ] torch.nn.modules.LazyLinear

# Dropout Layers
- [ ] torch.nn.modules.Dropout
- [ ] torch.nn.modules.Dropout2d
- [ ] torch.nn.modules.Dropout3d
- [ ] torch.nn.modules.AlphaDropout

# Sparse
- [ ] torch.nn.modules.Embedding
- [ ] torch.nn.modules.EmbeddingBad

# Distance Functions
- [ ] torch.nn.modules.CosineSimilarity
- [ ] torch.nn.PairwiseDistance

# Loss Functions
- [ ] torch.nn.modules.L1Loss
- [ ] torch.nn.modules.MSELoss
- [ ] torch.nn.modules.CrossEntropyLoss
- [ ] torch.nn.modules.CTCLoss
- [ ] torch.nn.modules.NLLLoss
- [ ] torch.nn.modules.PoissonNLLLos
- [ ] torch.nn.modules.GaussianNLLLoss
- [ ] torch.nn.modules.KLDivLoss
- [ ] torch.nn.modules.BCELoss
- [ ] torch.nn.modules.BCEWithLogitsLoss
- [ ] torch.nn.modules.MarginRankingLoss
- [ ] torch.nn.modules.HingeEmbeddingLoss
- [ ] torch.nn.modules.MultiLabelMarginLoss
- [ ] torch.nn.modules.HuberLoss
- [ ] torch.nn.modules.SmoothL1Loss
- [ ] torch.nn.modules.SoftMarginLoss
- [ ] torch.nn.modules.MultiLabelSoftMarginLoss
- [ ] torch.nn.modules.CosineEmbeddingLoss
- [ ] torch.nn.modules.MultiMarginLoss
- [ ] torch.nn.modules.TripletMarginLoss
- [ ] torch.nn.modules.TripletMarginWithDistanceLoss

# Vision Layers
- [ ] torch.nn.modules.PixelShuffle
- [ ] torch.nn.modules.PixelUnshuffle
- [ ] torch.nn.modules.Upsample
- [ ] torch.nn.modules.UpsamplingNearest2d
- [ ] torch.nn.modules.UpsamplingBilinear2d

# Shuffle Layers
- [ ] torch.nn.modules.ChannelShuffle
